<launch>
	<node name="odom_estimation_node" pkg="ukftest" type="odom_estimation_node"/>

	<param name="is_debug_on" value="true" />
	<node name="ASIRead" pkg="read_web_cam" type="ASIRead"/>
	<!-- Camera parameters -->
	<param name="set_gain" value="40" />
	<param name="set_exposure" value="60000" />
	<param name="set_height" value="240" />
	<param name="set_width" value="320" />
	<param name="set_bright" value="10" />
	
	<node name="aruco_tracker" pkg="irobot_tracker" type="aruco_tracker"/>
	<param name="markerSize" value="19.05" />  
	
	<node name="serial_to_uav_node" pkg="serial_to_uav" type="serial_to_uav_node" />
	<param name="is_write_on" value="true" />

	<node name="uav_tracking_controller_node" pkg="ukftest" type="uav_tracking_controller_node"/>
	<!-- PID parameters -->
	<param name="Kp_pos" value="0.90" />
	<param name="Ki_pos" value="0.10" />
	<param name="Kd_pos" value="0.005" />
	<param name="Kp_vel" value="2.50" />
	<param name="Ki_vel" value="0.10" />
	<param name="Kd_vel" value="0.10" />
	<param name="Kp_pos_y" value="0.9" />
	<param name="Ki_pos_y" value="0.1" />
	<param name="Kd_pos_y" value="0.01" />
	<param name="Kp_vel_y" value="2.5" />
	<param name="Ki_vel_y" value="0.10" />
	<param name="Kd_vel_y" value="0.34" />
        <param name="Kp_pos_z" value="0.0100" />
        <param name="Ki_pos_z" value="0.0001" />
        <param name="Kd_pos_z" value="0.0010" />
        <param name="Kp_yaw" value="0.800" />
        <param name="Ki_yaw" value="0.050" />
        <param name="Kd_yaw" value="1.000" />
	<param name="pid_gain" value="1.0" />
	<param name="controlLimit" value="5.0" />
	<param name="controlLimitVel" value="0.15" />
	<param name="controlLimitYawrate" value="10.0" />


	<node name="rplidarNode"          pkg="rplidar_ros"  type="rplidarNode">
	<param name="serial_port"         type="string" value="/dev/ttyUSB0"/>  
	<param name="serial_baudrate"     type="int"    value="115200"/>
	<param name="frame_id"            type="string" value="laser"/>
	<param name="inverted"            type="bool"   value="true"/>
	<param name="angle_compensate"    type="bool"   value="true"/>
	</node>

	<node name="my_vfh_4rp_node" pkg="ukftest" type="my_vfh_4rp_node" />
	
	<node name="uav_avoid_neg_controller_node" pkg="ukftest" type="uav_avoid_neg_controller_node" />
	<!-- Avoiding PID parameters -->
	<param name="Kp_avoid_vel" value="2.5" />
	<param name="Ki_avoid_vel" value="0.0" />
	<param name="Kd_avoid_vel" value="0.05" />
	<param name="pid_gain_avoid" value="1.0" />
	<param name="controlLimit_avoid" value="30.0" />


	<node name="uav_state_track_avoid_controller_node" pkg="ukftest" type="uav_state_track_avoid_controller_node" output="screen" />
	<!-- status parameters -->
	<param name="yaw_rate" value="0.0" />
	

</launch>

